src/backend/ai-reviewer-api/src/main/java/ca/bc/gov/open/jag/aireviewerapi/cso/CSOORDSServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package ca.bc.gov.open.jag.aireviewerapi.cso;
+
+import ca.bc.gov.open.jag.aireviewerapi.Keys;
+import ca.bc.gov.open.jag.aireviewerapi.api.model.ProcessedDocument;
+import ca.bc.gov.open.jag.aireviewerapi.cso.properties.CSOProperties;
+import ca.bc.gov.open.jag.aireviewerapi.extract.models.ExtractResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.text.MessageFormat;
+
+@Service
+@EnableConfigurationProperties({CSOProperties.class})
+public class CSOORDSServiceImpl implements CSOORDSService {
+
+    private final RestTemplate restTemplate;
+    private final CSOProperties CSOProperties;
+
+    Logger logger = LoggerFactory.getLogger(CSOORDSServiceImpl.class);
+
+    public CSOORDSServiceImpl(RestTemplate restTemplate, CSOProperties CSOProperties) {
+        this.restTemplate = restTemplate;
+        this.CSOProperties = CSOProperties;
+    }
+
+
+    @Override
+    public void sendExtractedData(ProcessedDocument processedDocument) {
+
+        int attempt = 0;
+        int maxAttempt = 5;
+        logger.info("Sending transaction {} extract to cso", processedDocument.getExtract().getTransactionId());
+
+        while (attempt < maxAttempt) {
+            logger.info("Attempting to send extract try number {}", (attempt + 1));
```

Comment:
Extraction Attempt number {}

Reply:
Or just start at 1 instead of 0, and go to attemp <= maxAttempt

Reply:
Yeah I have updated the logic.